modules/serverless-rn-1-17-0.adoc

```diff
@@ -0,0 +1,60 @@
+
+* Changes to API versions in Knative Eventing 0.23.0:
+
+** Version `v1beta1` has been removed from the `eventing.knative.dev`, `messaging.knative.dev` and `flows.knative.dev` APIs. You must ensure that any existing or new resources in your deployment that are created by using these APIs use version `v1`.
+
+** The `v1alpha1` version of the `KafkaChannel` API, which was deprecated in {ServerlessProductName} version 1.14.0, has been removed. If the `ChannelTemplateSpec` of your config maps contain references to this older version, you must update this part of the spec to use the correct API version.
+
+** The deprecated `v1alpha1`, `v1alpha2`, and `v1beta1` versions of the `APIServerSource` and `SinkBinding` APIs have been removed. You must update any existing resources to use the correct API versions.
+
+** The deprecated `v1alpha2` and `v1beta1` versions of the `ContainerSource` and `PingSource` APIs have been removed. You must update any existing resources to use the correct API versions.
+
+** Version `v1beta2` of the `PingSource` API has been deprecated. References to older versions of this API should be updated to `v1` to prepare your deployment for removal of the deprecated version.
+
+[id="fixed-issues-1-17-0_{context}"]
+== Fixed issues
+
+[id="known-issues-1-17-0_{context}"]
+== Known issues
+
+* If you try to use an older version of the Knative `kn` CLI with a newer {ServerlessProductName} release, the API is not found and an error occurs.
++
+For example, if you use the 1.16.0 release of the `kn` CLI, which uses version 0.22.0, with the 1.17.0 {ServerlessProductName} release, which uses the 0.23.0 versions of the Knative Serving and Knative Eventing APIs, the CLI does not work because it continues to look for the outdated 0.22.0 API versions.
++
+Ensure that you are using the latest `kn` CLI version for your {ServerlessProductName} release to avoid issues.
+
+* If you create a new subscription for a Kafka channel, there might be a delay in the Kafka channel data plane becoming ready to dispatch messages after the newly created subscription reports a ready status. As a result, messages that are sent during the time which the data plane is not reporting a ready status might not be delivered to the subscriber.
++
+The following steps are recommended to mitigate this issue:
+** Schedule a maintenance window for updates, during which the Kafka channel is not expected to receive events.
+** If you are creating subscriptions manually, send a few test events to ensure that events are not being lost. After you have confirmed that test events are not being lost, it is safe to continue using your subscriptions as usual.
+** If your subscriptions are being created by using automation, avoid using the current version of Kafka channels included with {ServerlessProductName} 1.17.0 for mission critical or time sensitive systems where some issues being lost when subscriptions are created is unacceptable.
```

Comment:
@devguyio if this is exactly the same but for source (https://issues.redhat.com/browse/SRVKE-911) I think we can just roll these into one issue; there's no need to duplicate all this, it would be easier to just say it applies for channels and sources. WDYT?

Reply:
```suggestion
* If you create a new subscription for a Kafka channel, there might be a delay in the Kafka channel data plane becoming ready to dispatch messages after the newly created subscription reports a ready status. As a result, messages that are sent during the time which the data plane is not reporting a ready status might not be delivered to the subscriber.
+
The following steps are recommended to mitigate this issue:
** Schedule a maintenance window for updates, during which the Kafka channel is not expected to receive events.
** If you are creating subscriptions manually, send a few test events to ensure that events are not being lost. After you have confirmed that test events are not being lost, it is safe to continue using your subscriptions as usual.
** If your subscriptions are being created by using automation and you're using the current version of Kafka channels included with {ServerlessProductName} 1.17.0 for mission critical or time sensitive systems where subscriptions missing events after creation is unacceptable, introduce an acknowledgement technique between the producer and subscribers (e.g. initial heartbeats) that signals the subscriber events recipition.
```

Reply:
> @devguyio if this is exactly the same but for source (https://issues.redhat.com/browse/SRVKE-911) I think we can just roll these into one issue; there's no need to duplicate all this, it would be easier to just say it applies for channels and sources. WDYT?


@abrennan89 yeah I think so too , would be awesome if you can find a way to say Source/Channel and Subscriber/Sink lol but I agree.

Reply:
Updated the suggestion again after the discussion below.

Reply:
@devguyio made some updates, lmk what you think, still reading through comments